### PR TITLE
Add inline documentation for JSON

### PR DESCRIPTION
**Pull request by gfontenot**

### CODE REVIEW

File: Argo/Types/JSON.swift [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import Foundation
 
+/// A type safe representation of JSON
```
Comment (edwardloveall) [inline Argo/Types/JSON.swift]: Three `/` ?

File: Argo/Types/JSON.swift [hunk 2]
```diff
@@ -9,6 +10,15 @@ public enum JSON {
 }
 
 public extension JSON {
+  /**
```
Comment (edwardloveall) [inline Argo/Types/JSON.swift]: Two `*`?

**Comment by edwardloveall** [comment]
Couple of comment syntax comments (which might be intentional? not sure). Otherwise, looks good to me.

File: Argo/Types/JSON.swift (hunk 1 above)
Comment (gfontenot) [inline Argo/Types/JSON.swift]: Single line doc comment

File: Argo/Types/JSON.swift (hunk 2 above)
Comment (gfontenot) [inline Argo/Types/JSON.swift]: Yep, `/*` is a multi-line comment, `/**` is a multi-line doc comment. This (and the `///` for single line comments) prompt Xcode to parse and display these blocks inside the IDE using a special flavor of markdown.

File: Argo/Types/JSON.swift [hunk 3]
```diff
@@ -21,6 +31,16 @@ public extension JSON {
 }
 
 extension JSON: Decodable {
+  /**
+    Decode `JSON` into `Decoded<JSON>`.
+
+    This simply wraps the provided `JSON` in `Success`. This is useful because
```
Comment (tonyd256) [inline Argo/Types/JSON.swift]: `.Success`

**Comment by tonyd256** [comment]
1 typo otherwise :+1:

File: Argo/Types/JSON.swift (hunk 1 above)
Comment (tonyd256) [inline Argo/Types/JSON.swift]: missing period

File: Argo/Types/JSON.swift [hunk 4]
```diff
@@ -9,6 +10,15 @@ public enum JSON {
...
 
 public extension JSON {
+  /**
+    Transform an `AnyObject` instance into `JSON`.
+
+    This is used to move from a loosely typed object (like those returned from
+    `NSJSONSerialization`) to the strongly typed `JSON` tree structure.
+
+    - parameter json: A loosely typed object
+    - returns: An instance of the JSON tree structure
```
Comment (tonyd256) [inline Argo/Types/JSON.swift]: extra line needed

File: Argo/Types/JSON.swift [hunk 5]
```diff
@@ -21,6 +31,16 @@ public extension JSON {
...
 extension JSON: Decodable {
+  /**
+    Decode `JSON` into `Decoded<JSON>`.
+
+    This simply wraps the provided `JSON` in `Success`. This is useful because
+    it means we can use `JSON` values with the `<|` family of operators to pull
+    out sub-keys.
+
+    - parameter j: The `JSON` value to decode
+    - returns: The provided `JSON` wrapped in `Success`
```
Comment (tonyd256) [inline Argo/Types/JSON.swift]: new line above

**Pull request merged**